clouddriver-aws/src/main/groovy/com/netflix/spinnaker/clouddriver/aws/deploy/ops/loadbalancer/UpsertAmazonLoadBalancerAtomicOperation.groovy

```diff
@@ -192,62 +191,88 @@ class UpsertAmazonLoadBalancerAtomicOperation implements AtomicOperation<UpsertA
+                                                            SecurityGroupLookupFactory securityGroupLookupFactory) throws FailedSecurityGroupIngressException {
+    String application = Names.parseName(description.name).getApp() ?: Names.parseName(description.clusterName).getApp()
     SecurityGroupLookup securityGroupLookup = securityGroupLookupFactory.getInstance(region)
-    String applicationSecurityGroupName = description.application
-    String applicationLoadBalancerSecurityGroupName = description.application + "-elb"
+    String applicationLoadBalancerSecurityGroupName = application + "-elb"
 
     // 1. get app load balancer security group & app security group. create if doesn't exist
-    String applicationLoadBalancerSecurityGroupId = getOrCreateSecurityGroup(
+    SecurityGroupUpdater applicationLoadBalancerSecurityGroupUpdater = getOrCreateSecurityGroup(
       applicationLoadBalancerSecurityGroupName,
       description.vpcId,
       description.credentialAccount,
       region,
-      "Application ELB Security Group for $description.application",
+      "Application ELB Security Group for $application",
       securityGroupLookup
     )
 
-    String applicationSecurityGroupId = getOrCreateSecurityGroup(
-      applicationSecurityGroupName,
+    SecurityGroupUpdater applicationSecurityGroupUpdater = getOrCreateSecurityGroup(
+      application,
       description.vpcId,
       description.credentialAccount,
       region,
-      "Application Security Group for $description.application",
+      "Application Security Group for $application",
       securityGroupLookup
     )
 
-    IngressSecurityGroupDescription applicationSecurityGroupDescription = new IngressSecurityGroupDescription(
-      applicationSecurityGroupId,
-      description.name,
-      description.credentialAccount,
-      description.vpcId
-    ).withHealthCheck(description.healthCheck).withListeners(loadBalancerListeners)
+    def source = applicationLoadBalancerSecurityGroupUpdater.securityGroup
+    def target = applicationSecurityGroupUpdater.securityGroup
+    List<IpPermission> currentPermissions = SecurityGroupIngressConverter.flattenPermissions(target)
+    List<IpPermission> targetPermissions = loadBalancerListeners.collect {
+      new IpPermission(
+        ipProtocol: "tcp",
+        fromPort: it.getInstancePort(),
+        toPort: it.getInstancePort(),
+        userIdGroupPairs: [
+          new UserIdGroupPair().withGroupId(source.groupId)
+        ]
+      )
+    }
 
-    IngressSecurityGroupDescription applicationLoadBalancerSecurityGroupDescription = new IngressSecurityGroupDescription(
-      applicationLoadBalancerSecurityGroupId,
-      applicationLoadBalancerSecurityGroupName,
-      description.credentialAccount,
-      description.vpcId
-    )
+    // include health check endpoint
+    Matcher portMatcher = HEALTHCHECK_PORT_PATTERN.matcher(description.healthCheck)
```

Comment:
Is `description.healthCheck` the right attribute to check?

```
        "healthCheck": "HTTP:7001/healthcheck",
        "healthCheckProtocol": "HTTP",
        "healthCheckPort": 7001,
```

Could we just check `healthCheckPort`? Alternatively if we're checking `.healthCheck`, is there any implication if the value was `HTTP:7001/v2/healthcheck`?

Reply:
For simplicity, I will use `healthCheckPort` here